packages/backend/src/nest/storage/storage.service.ts

```diff
@@ -419,6 +423,25 @@ export class StorageService extends EventEmitter {
     this.logger('STORAGE: Finished creating certificatesRequests db')
   }
 
+  public async filterCsrs(allCsrs: string[]): Promise<string[]> {
+    const filteredCsrsMap: Map<string, string> = new Map()
+
+    await Promise.all(
+      allCsrs.map(async csr => {
+        const parsedCsr = await loadCSR(csr)
+        const pubKey = keyFromCertificate(parsedCsr)
+
+        if (filteredCsrsMap.has(pubKey)) {
+          filteredCsrsMap.delete(pubKey)
+        }
```

Comment:
I don't think there is any need for  this, just ```filteredCsrsMap.set(pubKey, csr)```

Reply:
This is quite important. 

We want to have correct order in that data structure when user didn't immediately change his name, because he can use app with duplicated name.

For example map of users pubkey=> csr(and i added nickname)

`{"pubkey1" => "csr1/owner", "pubkey2" => "csr2/alice", "pubkey3" => "csr3/alice(duplicated)", "pubkey4" => "csr4/jimmy"}`


Case that we dont delete existed key but replace it  will be this: 
Example alice(duplicated) change name after `jimmy` joined to community to `jimmy`

`{"pubkey1" => "csr1/owner", "pubkey2" => "csr2/alice", "pubkey3" => "jimmy", "pubkey4" => "csr4/jimmy(duplicated)"}`


But we want something like this:

`{"pubkey1" => "csr1/owner", "pubkey2" => "csr2/alice", "pubkey4" => "csr4/jimmy", "pubkey3" => "jimmy(duplicated)"}`

Reply:
Ok, got it, so the insertion order is based on when the key was originally inserted. That makes sense, nice catch!